manifests/base.pp

```diff
@@ -44,27 +36,17 @@
   }
 
   if $rsyslog::manage_confdir {
-    $purge_params = $rsyslog::purge_config_files ? {
-      true  => {
-        'purge'   => true,
-        'recurse' => true,
-      },
-      false => {}
-    }
-
-    $require_package = $rsyslog::manage_package ? {
-      true => {
-        'require' => Package[$rsyslog::package_name],
-      },
-      false => {}
+    file { $rsyslog::confdir:
+      ensure  => directory,
+      owner   => 'root',
+      group   => 'root',
+      mode    => $rsyslog::confdir_permissions,
+      purge   => $rsyslog::purge_config_files,
```

Comment:
Note `purge` and `recurse` both default to `false` in file resources, so this shouldn't change behaviour.